xacro/__init__.py

```diff
@@ -1018,7 +1022,8 @@ def parse(inp, filename=None):
     f = None
     if inp is None:
         try:
-            inp = f = open(filename)
+            global root_dir
```

Comment:
You are just reading `root_dir`. There is no need to declare the variable global here, right?
```suggestion
```